src/pages/MainPage.vue

```diff
@@ -204,31 +159,23 @@
         ref="conflictdialog"
         :conflict-info="conflictInfo"
         :conflict-dialog="conflictDialog"
-        :roads="roads"
         @update-local="updateLocal"
         @update-remote="updateRemote"
       />
     </v-content>
 
     <class-info
-      v-if="classInfoStack.length"
-      :class-info-stack="classInfoStack"
-      :subjects="subjectsInfo"
-      :subjects-index="subjectsIndexDict"
+      v-if="$store.state.classInfoStack.length"
       :adding-from-card="addingFromCard"
-      :generic-courses="genericCourses"
-      :generic-index="genericIndexDict"
-      @pop-stack="popClassStack"
-      @push-stack="pushClassStack"
       @add-class="addFromCard"
       @cancel-add-class="cancelAddFromCard"
-      @close-classinfo="classInfoStack = []"
+      @close-classinfo="$store.commit('clearClassInfoStack')"
```

Comment:
For things like this, should we move them down into the component now that we have Vuex or did you leave it this way for easier readability and since this is a huge PR already? I think that's a good argument but also it seems like other things have been moved, so we should probably do all or none for consistency.

Reply:
That's just something I missed. It definitely should be moved down into the component. There are probably other things that I also missed.